app/controllers/warehouse_reports/missing_values_controller.rb

```diff
@@ -216,10 +216,13 @@ def all_sources
           st = data_sources.arel_table
           ot = organizations.arel_table
           pt = projects.arel_table
-          sql = data_sources.importable.joins(organizations: :projects).
-            merge(projects.viewable_by(user)).
-            select(st[:id], st[:name], st[:short_name], ot[:id], ot[:OrganizationName], pt[:id], pt[:ProjectName]).
-            to_sql
+
+          scope = data_sources.importable.joins(organizations: :projects).merge(projects.viewable_by(user))
+          scope = scope.merge(organizations.non_confidential) unless user.can_view_confidential_enrollment_details?
+          scope = scope.merge(projects.non_confidential) unless user.can_view_confidential_enrollment_details?
```

Comment:
This is really minor, but maybe:
```suggestion
          scope = data_sources.importable.joins(organizations: :projects).merge(projects.viewable_by(user))
          unless user.can_view_confidential_enrollment_details?
            scope = scope.merge(organizations.non_confidential).
              merge(projects.non_confidential) 
          end
```